### PR TITLE
Use embedded-hal 1.0.0

### PR DESCRIPTION
**Pull request by dan-corneanu**
Update lib to use embedded-hal 1.0.0

### CODE REVIEW

**Comment by dan-corneanu** [comment]
@eldruin I've updated the driver to use embedded-hal 1.0.0 because I have a project in which I need to share the same SPI bus between two devices. However, it seems that the specs fail on some of the architectures and I have no clue what is wrong. Could you please advise?

File: Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,8 @@ include = [
 edition = "2018"
 
 [dependencies]
-embedded-hal = "0.2.6"
+embedded-hal = { version = "1.0.0"}
+embedded-hal-bus = { version = "0.2.0" }
```
Comment (eldruin) [inline Cargo.toml]:
```suggestion
```
We do not need this in a driver

File: Cargo.toml [hunk 2]
```diff
@@ -21,7 +21,8 @@ include = [
...
 
 [dev-dependencies]
 linux-embedded-hal = "0.3"
```
Comment (eldruin) [inline Cargo.toml]:
```suggestion
linux-embedded-hal = "0.4"
```

**Review by eldruin — COMMENTED** [review]
Thank you for your work!
The problem in the CI is because of the `embedded-hal-bus` dependency but we do not need it here.
The tests and examples would need to be adapted as well, though.
For that you can update the `embedded-hal-mock` dependency to `embedded-hal-mock = { version="0.10", default-features=false, features=["eh1"]}`
You can have a look at this PR for inspiration, for example: https://github.com/eldruin/bmi160-rs/pull/7/files

**Comment by dan-corneanu** [comment]
@eldruin I've updated the tests and all of them are passing now. However, I don't know how to solve the `cargo-tarpaulin` and `clippy` issues. I need your help with these two.

**Comment by eldruin** [comment]
Thank you! The CI configuration was a bit outdated. I have updated it and now everything passes except one trivial error reported by clippy.
Could you also add an entry to the changelog?

**Comment by dan-corneanu** [comment]
Would you mind giving me your advice on https://github.com/rust-embedded/embedded-hal/issues/608 please?

**Review by eldruin — COMMENTED** [review]
Great, thank you.
Sorry, could you also update the example in the readme?

**Review by eldruin — APPROVED** [review]
Great work, thank you!

**Comment by eldruin** [comment]
For the record, I have published this in [version 1.0.0](https://github.com/eldruin/mcp4x-rs/releases/tag/v1.0.0).

**Pull request merged**